bitcoin/src/bip32.rs

```diff
@@ -31,11 +31,11 @@ const VERSION_BYTES_TESTNETS_PRIVATE: [u8; 4] = [0x04, 0x35, 0x83, 0x94];
 
 /// The old name for xpub, extended public key.
 #[deprecated(since = "0.31.0", note = "use xpub instead")]
-pub type ExtendendPubKey = Xpub;
+pub type ExtendedPubKey = Xpub;
```

Comment:
Interesting, we should just delete this code. The type was named `ExtendedPubKey` in 0.30 and we removed `ExtendedPubKey` completely in 0.31 by accidentally mis-spelling it.

Reply:
Yes, I also agree.

Reply:
Agreed :(. We messed up the deprecation cycle here. Frustrating that nobody noticed until now -- it would have made the 0.30 to 0.31 transition much easier.

Maybe this PR should instead be a backport to the 0.31 branch.

Also the deprecation method should say `Xpub` instead of xpub, which will make it clear that there's an actual type called `Xpub`.

Reply:
> Note that with the CI testing, it took a long time for someone to find this typo.

I'm not sure that tests would have caught it. Tobin and I updated a bunch of crates from 0.30 to 0.31, and I recall seeing that `ExtendedPubKey` no longer compiled and thinking "hmm don't I remember us doing a deprecation here?" and then deciding that there must've been some reason we didn't.

Reply:
@tcharding @apoelstra the deprecation warning does not show up unless you explicitly wrap the create a new type. Simply `pub type A = B` does an alias and would not show warning when user uses type A.

Reply:
@tcharding @apoelstra the deprecation warning does not show up unless you explicitly wrap the create a new type. Simply `pub type A = B` does an alias and would not show warning when user uses type A.

Reply:
Perhaps we should do as you suggested once in the past @apoelstra and during RC testing I should do each upgrade as two separate patches, the first being required changes and the second fixing all the lints (deprecation). That way we may notice if something is in the first patch that shouldn't have been.

Reply:
Agreed, though if Sanket is correct, we won't actually get a deprecation warning in this case :(.

Reply:
I swapped `Xpriv` for `ExtendendPrivKey` on `master` (with the spelling mistake) and got a warning
```
cargo run --example taproot-psbt --features=rand-std,bitcoinconsensus
warning: /home/tobin/build/github.com/tcharding/rust-bitcoin/master/bitcoin/Cargo.toml: version requirement `0.105.0+25.1` for dependency `bitcoinconsensus` includes semver metadata which will be ignored, removing the metadata is recommended to avoid confusion
   Compiling bitcoin v0.32.0-rc1 (/home/tobin/build/github.com/tcharding/rust-bitcoin/master/bitcoin)
warning: use of deprecated type alias `bitcoin::bip32::ExtendendPrivKey`: use xpriv instead
  --> bitcoin/examples/taproot-psbt.rs:81:77
   |
81 | use bitcoin::bip32::{ChildNumber, DerivationPath, Fingerprint, Xpriv, Xpub, ExtendendPrivKey};
   |                                                                             ^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(deprecated)]` on by default

warning: use of deprecated type alias `bitcoin::bip32::ExtendendPrivKey`: use xpriv instead
   --> bitcoin/examples/taproot-psbt.rs:117:9
    |
117 |         ExtendendPrivKey::from_str(BENEFACTOR_XPRIV_STR)?,
    |         ^^^^^^^^^^^^^^^^
```

Reply:
Thanks for checking. For some reason, I recall this not working.

Reply:
No sweat, there are various deprecation things that definitely do not work as they appear, I can not remember which ones work and which don't either.